torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -265,6 +265,9 @@ void PropagateShapeOnNode(Node * node, bool insert_expands) {
   }
   if (node->matches("aten::cat(Tensor[] tensors, int dim) -> Tensor", /*with_const=*/attr::dim)) {
     auto list_node = node->namedInput(attr::tensors)->node();
+    if (list_node->kind() == prim::Loop || list_node->kind() == prim::If) {
+      goto cat_fail;;
+    }
```

Comment:
I don't think this is the right fix. We should just remove the `JIT_ASSERT` and simply `goto cat_fail` in any other case. It's not like we will know the length of the list unless it's actually a constant.

Reply:
I didn't intend this as a permanent fix, just as something to stop it from throwing an exception.